### DISCUSSION TITLE
generating a star shaped domain with a point cloud inside it

### DISCUSSION THREAD

**Discussion by reyhanhgh** [yalmip/YALMIP#1151] [Problems using YALMIP]
Hi!
I need to generate a star shaped domain (x-y coordinates) centered around (`x_0, y_0`) of width `W` with `N` random points inside it in python. Conforming to the matlab code provided in [YALMIP star convex polygon](https://yalmip.github.io/example/starshaped/), I managed to reproduce this part :

```
n = 6;
th = linspace(-pi, pi, 2*n+1);
xi = (1+rem((1:2*n+1),2)).*cos(th);
yi = (1+rem((1:2*n+1),2)).*sin(th);
clf;
hold on;
grid on
plot( xi, yi, 'b*-' );
axis equal;
```
I have two questions/problems:

Question 1- This gives a star centered at (0,0) and of width 4. But it is not clear to me how I can control these parameters in the above snippet code.

Question 2- My second problem regards the generation of the N random point in the star. The snippet code provided by YALMIP for  [Optimizing over star-convex](https://yalmip.github.io/example/starshaped/#:~:text=Optimizing%20over%20star%2Dconvex%20polygons) seems related to what I am looking for. But, again I can not figure out how can I control the number of points inside the star, and most importantly how I can generate them without having the need to translate them.

Your help would be highly appreciated.

**Comment by johanlofberg** [comment]
To translate and scale, well just add the offset and scale xi and yi

To create a random point, one way would be
```
s = -pi+rand(1)*2*pi;
r = rand(1);
x = r*interp1(th,xi,s);
y = r*interp1(th,yi,s);
hold on;
grid on
plot( x, y, 'r*' );
axis equal;
```

This is probably not uniformly distributed in the star though, to figure out how one would sample r and s for that is another issue

**Comment by reyhanhgh** [comment]
Thanks Johan. For generation of uniform random points I used the solution proposed [here](https://stackoverflow.com/questions/55392019/get-random-points-within-polygon-corners#:~:text=In%20Python%20you%20can%20generate%20random%20coordinates%20with%20in%20a%20defined%20polygon%20like%20below).
